.github/workflows/types.yaml

```diff
@@ -10,111 +10,171 @@ jobs:
       - name: Generate ABI
         run: yarn abi:generate
 
-      - name: Generate types bindings
+      # Step 7: Generate TypeScript bindings from ABI
+      - name: Generate TypeScript Bindings
         run: yarn typechain
 
-      - name: Checkout lifi-contract-types repository
-        uses: actions/checkout@v4
+      # Step 8: Checkout the lifi-contract-types repository
+      - name: Checkout lifi-contract-types Repository
+        uses: actions/checkout@v4.1.7
         with:
           repository: lifinance/lifi-contract-types
           path: lifi-contract-types
           ssh-key: ${{ secrets.SSH_REPO_TOKEN }}
           ref: main
 
-      - name: Copy types bindings
+      # Step 9: Copy generated types and ABI into the lifi-contract-types repo
+      - name: Copy Type Bindings and ABI
         run: |
           rm -r lifi-contract-types/src/
           mv typechain lifi-contract-types/src
           cp diamond.json lifi-contract-types/dist/
 
-      - name: Build contract types
+      # Step 10: Build the lifi-contract-types project
+      - name: Build Contract Types
         run: cd lifi-contract-types && yarn install && yarn build
 
-      - name: Retrieve latest Tag
+      # Step 11: Retrieve the latest tag from the repository
+      - name: Retrieve Latest Tag
         id: latest_release
         run: |
-          # fetch tag releases
-          release_json=$(curl https://api.github.com/repos/lifinance/lifi-contract-types/tags)
+          # Fetch tags from the lifi-contract-types repository
+          RELEASE_JSON=$(curl https://api.github.com/repos/lifinance/lifi-contract-types/tags)
 
-          # get the latest tag
-          LATEST_TAG=$(echo "$release_json" | jq -r '.[0].name')
+          # Extract the latest tag name
+          LATEST_TAG=$(echo "$RELEASE_JSON" | jq -r '.[0].name')
 
-          # we need to make sure that on staging we're going to update a -beta version, if any
+          # Handle beta versions for non-main branches
           if [[ "$BRANCH_NAME" != "main" ]]; then
-            # if it has already "-beta", no other action is required, since it means
-            # that we're already going to update the latest staging release
             if [[ "$LATEST_TAG" != *"beta"* ]]; then
-              # otherwise, start looping through the tags and search for the latest -beta tag
-              while read item; do
-                tag_name=$(jq -r '.name' <<< "$item")
-                # check if there's already a latest tag beta release
-                # and, if present, use it instead of the main one
-                # if we end up without any latest beta tag, we will create a beta release from the latest tag
-                if [[ "$tag_name" == "$LATEST_TAG-$beta"* ]]; then
-                  LATEST_TAG=$tag_name
+              while read ITEM; do
+                TAG_NAME=$(jq -r '.name' <<< "$ITEM")
+                if [[ "$TAG_NAME" == "$LATEST_TAG-$beta"* ]]; then
```

Comment:
Dont know if you mean to but $beta is in lowercase here.

Reply:
works like this...seems fine